nodejs/aws-infra/iam/index.ts

```diff
@@ -0,0 +1,213 @@
+import * as aws from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+
+import * as utils from "./../utils";
+
+export class User extends pulumi.ComponentResource {
+    public readonly user: aws.iam.User;
+    public readonly groupMembership?: aws.iam.UserGroupMembership;
+    public readonly accessKey?: aws.iam.AccessKey;
+
+    constructor(name: string, args: UserArgs = {}, opts?: pulumi.CustomResourceOptions) {
+        super(`awsinfra:x:iam:User`, name, args, opts);
+
+        // Save group membership and access key args so we can define them inline.
+        const groupMembership = args.groupMembership;
+        const accessKey = args.accessKey;
+
+        // Explicitly delete these props so we do *not* pass them into the User created
+        // below.
+        delete args.groupMembership;
+        delete args.accessKey;
+        delete args.user;
+
+        this.user = args.user || new aws.iam.User(name, args && args, { parent: this });
+
+        if (groupMembership !== undefined) {
+            this.groupMembership = new aws.iam.UserGroupMembership(
+                name,
+                { ...groupMembership, user: this.user.name },
+                { parent: this }
+            );
+        }
+
+        if (accessKey == "create") {
+            this.accessKey = new aws.iam.AccessKey(
+                name,
+                { user: this.user.name },
+                { parent: this }
+            );
+        } else if (args && accessKey !== undefined && accessKey != "create") {
+            this.accessKey = new aws.iam.AccessKey(
+                name,
+                { ...accessKey, user: this.user.name },
+                { parent: this }
+            );
+        }
+    }
```

Comment:
i'm on the fence about hte accesskey/group stuff being in the constructor.  perhaps it would be nicer to model as instance methods like:

`addToGroup(...)` and `createAccessKey(...)`?

Reply:
I believe:

* The vast majority of human `User`s will be added to groups, and therefore I think it makes sense to include that in the constructor.
* It will also be useful to allow people to add group memberships after instantiation, so an `addToGroups` function makes sense to me.

`AccessKey` is not something most users will want, so I'm ok moving that to a method.

Happy to hear alternative views.

Reply:
A question I do have is whether `AccessKey` and `UserGroupMemberships` should be stored inside the `User` when using these methods. I see that `SecurityGroup` does not keep these resources, and instead returns them. I think that is confusing, but I'm guessing we had good reasons for doing that.